### PR TITLE
Multithread support

### PR DESCRIPTION
**Pull request by PoulpiFr**
All Plugins will be multi threaded by default

New PR, a previous version of it was: https://github.com/MEDIARITHMICS/plugins-nodejs-sdk/pull/18.

Unfortunately, some GitHub configuration changes (switch from private repo to a public one for the Plugin SDK) broke the previous PR.

### CODE REVIEW

File: src/core/class/mediarithmics/common/ProductionPluginRunner.ts [hunk 1]
```diff
@@ -1,27 +1,193 @@
...
+   */
+  masterListener = (worker: cluster.Worker, recMsg: SocketMsg) => {
+    this.plugin.logger.debug(
+      `Master ${process.pid} is being called with cmd: ${
+        MsgCmd[recMsg.cmd]
+      }, value: ${recMsg.value}`
+    );
+
+    // If we receive a Token Update, we update the token
+    if (recMsg.cmd === MsgCmd.CREDENTIAL_UPDATE_FROM_WORKER) {
```
Comment (thomastosoni) [inline src/core/class/mediarithmics/common/ProductionPluginRunner.ts]: You can merge the same `ifs` and else if the credentials and logs

Comment (thomastosoni) [reply src/core/class/mediarithmics/common/ProductionPluginRunner.ts]: You can also factorize the workers notifications and put the different processings depending on MsgCmd in separate methods.

File: src/core/class/mediarithmics/common/ProductionPluginRunner.ts [hunk 2]
```diff
@@ -1,27 +1,193 @@
...
+    // If we receive a Token Update, we propagate it to all workers
+    if (recMsg.cmd === MsgCmd.CREDENTIAL_UPDATE_FROM_MASTER) {
+      if (!recMsg.value) {
+        throw new Error(
+          "We received a CREDENTIAL_UPDATE_FROM_MASTER msg without credentials in the value field of the msg."
+        );
+      }
+      const creds: Credentials = JSON.parse(recMsg.value);
+
+      const initUpdateResult = this.plugin.onInitRequest(creds);
```
Comment (thomastosoni) [inline src/core/class/mediarithmics/common/ProductionPluginRunner.ts]: Remove all 3 unused constants

**Comment by PoulpiFr** [comment]
@thomastosoni > Can you check again?

**Comment by PoulpiFr** [comment]
@thomastosoni > ping

As we're preparing the v0.4 of the SDK + as some Audience External Feed have some performance issues, merging the PR this week / beginning of next week would be perfect =)

**Comment by PoulpiFr** [comment]
@thomastosoni > ping

**Pull request merged**